schemas/JSON/manifests/v1.2.0/manifest.installer.1.2.0.json

```diff
@@ -0,0 +1,672 @@
+      "items": {
+        "type": "object",
+        "title": "ExpectedReturnCode",
+        "properties": {
+          "InstallerReturnCode": {
+            "$ref": "#/definitions/InstallerReturnCode"
+          },
+          "ReturnResponse": {
+            "type": "string",
+            "enum": [
+              "packageInUse",
+              "installInProgress",
+              "fileInUse",
+              "missingDependency",
+              "diskFull",
+              "insufficientMemory",
+              "noNetwork",
+              "contactSupport",
+              "rebootRequiredToFinish",
+              "rebootRequiredForInstall",
+              "rebootInitiated",
+              "cancelledByUser",
+              "alreadyInstalled",
+              "downgrade",
+              "blockedByPolicy"
+            ]
+          }
+        }
+      },
+      "maxItems": 128,
+      "description": "Installer exit codes for common errors"
+    },
+    "UpgradeBehavior": {
+      "type": [ "string", "null" ],
+      "enum": [
+        "install",
+        "uninstallPrevious"
+      ],
+      "description": "The upgrade method"
+    },
+    "Commands": {
+      "type": [ "array", "null" ],
+      "items": {
+        "type": "string",
+        "minLength": 1,
+        "maxLength": 40
+      },
+      "maxItems": 16,
+      "uniqueItems": true,
+      "description": "List of commands or aliases to run the package"
+    },
+    "Protocols": {
+      "type": [ "array", "null" ],
+      "items": {
+        "type": "string",
+        "pattern": "^[a-z][-a-z0-9\\.\\+]*$",
+        "maxLength": 2048
+      },
+      "maxItems": 16,
+      "uniqueItems": true,
+      "description": "List of protocols the package provides a handler for"
+    },
+    "FileExtensions": {
+      "type": [ "array", "null" ],
+      "items": {
+        "type": "string",
+        "pattern": "^[^\\\\/:\\*\\?\"<>\\|\\x01-\\x1f]+$",
+        "maxLength": 64
+      },
+      "maxItems": 256,
+      "uniqueItems": true,
+      "description": "List of file extensions the package could support"
+    },
+    "Dependencies": {
+      "type": [ "object", "null" ],
+      "properties": {
+        "WindowsFeatures": {
+          "type": [ "array", "null" ],
+          "items": {
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 128
+          },
+          "maxItems": 16,
+          "uniqueItems": true,
+          "description": "List of Windows feature dependencies"
+        },
+        "WindowsLibraries": {
+          "type": [ "array", "null" ],
+          "items": {
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 128
+          },
+          "maxItems": 16,
+          "uniqueItems": true,
+          "description": "List of Windows library dependencies"
+        },
+        "PackageDependencies": {
+          "type": [ "array", "null" ],
+          "items": {
+            "type": "object",
+            "properties": {

```

Comment:
I know this is the "initial" schema, but any chance we could get the architecture key for PackageDependencies here per https://github.com/microsoft/winget-cli/issues/1665?

Reply:
@jedieaston I've run into a couple of cases where the dependencies aren't "natively" available for the architecture of the package. What would the intent be here? An optional "dependencyArchitecture" where the dependency needs to be other than the architecture of the installer? That's my best guess from reading #1665 and what I think I ran into trying to get Screen To Gif installed on ARM64.

Reply:
That would be my thinking. Many times installers are one architecture, but they rely on dependencies for a different architecture. I've seen lots of applications where they are x64 but for whatever reason they require x86 versions of the VC Redists.